src/pkg/services/m365/api/lists.go

```diff
@@ -0,0 +1,64 @@
+package api
+
+import (
+	"context"
+
+	"github.com/alcionai/clues"
+	"github.com/microsoftgraph/msgraph-sdk-go/models"
+
+	"github.com/alcionai/corso/src/internal/common/ptr"
+	"github.com/alcionai/corso/src/internal/m365/graph"
+)
+
+// ---------------------------------------------------------------------------
+// controller
+// ---------------------------------------------------------------------------
+
+func (c Client) Lists() Lists {
+	return Lists{c}
+}
+
+// Lists is an interface-compliant provider of the client.
+type Lists struct {
+	Client
+}
+
+// PostDrive creates a new list of type drive.  Specifically used to create
+// documentLibraries for SharePoint Sites.
+func (c Lists) PostDrive(
+	ctx context.Context,
+	siteID, driveName string,
+) (models.Driveable, error) {
+	list := models.NewList()
+	list.SetDisplayName(&driveName)
+	list.SetDescription(ptr.To("corso auto-generated restore destination"))
```

Comment:
Just a question for the future. Should we have the name in here or provide a way to customize the name here in case SDK consumer wants change the description. For example to "Ark auto-generated restore destination".

Reply:
Good question.  I was going to ping Georgi about that later.  For now I think we're safe following standard folder naming conventions.  Though, now that I think about it, that doesn't allow SDK customers to override the corso naming conventions.  I'll probably do a follow-up PR where this gets added to the RestoreConfig.